src/id3/v2/tag.rs

```diff
@@ -266,16 +271,11 @@ impl ID3v2Tag {
 		})
 	}
 
-	/// Returns all `COMM` frames
+	/// Returns all `COMM` frames with an empty content descriptor
```

Comment:
Why should this be restricted to empty descriptors only? An ID3v2 tag shouldn't have any comments with duplicate descriptors according to spec anyway. It's definitely possible to create them and they will be written, but I don't think that should be encouraged.

Reply:
Those custom comments with non-empty content descriptors cause lots of issues. Think about all the possible round trips and you will notice how they cause inconsistencies and unexpected behavior.

If you are not able to distinguish them the restrict access to those with an empty content descriptor. Experience from Mixxx development and from seeing lots of different files in the wild.

TagLib: <https://github.com/taglib/taglib/blob/a31356e330674640a07bef7d71d08242cae8e9bf/taglib/mpeg/id3v2/id3v2tag.cpp#L175>
Mixxx: <https://github.com/mixxxdj/mixxx/blob/cce89d09f3a546f54af2c2d473a848f9cf6eee26/src/track/taglib/trackmetadata_id3v2.cpp#L697>

...and probably many others also.

Reply:
TagLib picks the first non-empty content descriptor as a fallback, which is bad because it can cause unexpected behavior. After writing the comment the original frame with the non-empty content descriptor remains in the file, probably even as the first one. Depending on whether other applications are handling the distinction by content descriptor correctly or not users will complain about comments not being written... Have fun.

Being strict and consistent is the only chance to avoid chaos.

Reply:
Thanks, I see how that could be an issue.